src/test/java/org/broadinstitute/hellbender/tools/copynumber/PreprocessIntervalsIntegrationTest.java

```diff
@@ -0,0 +1,146 @@
+        final int binLengthWholeGenomeTest = 10_000_000;
+        final int paddingWholeGenomeTest = 500;
+        final String[] outputFileNameWholeGenomeTest =  {"GATK-preprocess-intervals-test", ".tmp"};
+        final ArrayList<Interval> inputIntervalsWholeGenomeTest = new ArrayList<Interval>();
+        final ArrayList<Interval> expectedBinsWholeGenomeTest = new ArrayList<Interval>();
+        expectedBinsWholeGenomeTest.add(new Interval("20", 1, 10_000_000));
+        expectedBinsWholeGenomeTest.add(new Interval("20", 10_000_001, 20_000_000));
+        expectedBinsWholeGenomeTest.add(new Interval("20", 20_000_001, 30_000_000));
+        expectedBinsWholeGenomeTest.add(new Interval("20", 30_000_001, 40_000_000));
+        expectedBinsWholeGenomeTest.add(new Interval("20", 40_000_001, 50_000_000));
+        expectedBinsWholeGenomeTest.add(new Interval("20", 50_000_001, 60_000_000));
+        expectedBinsWholeGenomeTest.add(new Interval("20", 60_000_001, 63_025_520));
+        expectedBinsWholeGenomeTest.add(new Interval("21", 1, 10_000_000));
+        expectedBinsWholeGenomeTest.add(new Interval("21", 10_000_001, 20_000_000));
+        expectedBinsWholeGenomeTest.add(new Interval("21", 20_000_001, 30_000_000));
+        expectedBinsWholeGenomeTest.add(new Interval("21", 30_000_001, 40_000_000));
+        expectedBinsWholeGenomeTest.add(new Interval("21", 40_000_001, 48_129_895));
+
+        // Return all test data
+        return new Object[][]{
+                {binLengthSeparateIntervalTest, paddingSeparateIntervalTest, outputFileNameSeparateIntervalTest, inputIntervalsSeparateIntervalTest, expectedBinsSeparateIntervalTest},
+                {binLengthOverlappingIntervalTest, paddingOverlappingIntervalTest, outputFileNameOverlappingIntervalTest, inputIntervalsOverlappingIntervalTest, expectedBinsOverlappingIntervalTest},
+                {binLengthEdgeIntervalTest, paddingEdgeIntervalTest, outputFileNameEdgeIntervalTest, inputIntervalsEdgeIntervalTest, expectedBinsEdgeIntervalTest},
+                {binLengthWholeChromosomeTest, paddingWholeChromosomeTest, outputFileNameWholeChromosomeTest, inputIntervalsWholeChromosomeTest, expectedBinsWholeChromosomeTest},
+                {binLengthWholeGenomeTest, paddingWholeGenomeTest, outputFileNameWholeGenomeTest, inputIntervalsWholeGenomeTest, expectedBinsWholeGenomeTest}
+        };
+    }
+
+    // Test for interval inputs given as -L command line arguments
+    @Test(dataProvider = "intervalInputsFromCommandLine")
+    public final void testCommandLine(final int binLength, final int padding,  final String[] outputFileName, final ArrayList<Interval> inputIntervals, final ArrayList<Interval> expectedBins) {
+        final File outputFile = createTempFile(outputFileName[0], outputFileName[1]);
+        final ArrayList<String> args = new ArrayList<>(Arrays.asList(
```

Comment:
It may be easier to use an `ArgumentsBuilder` here, e.g.:

    final ArgumentsBuilder argsBuilder = new ArgumentsBuilder()
        .addReference(REFERENCE_FILE)
        .addArgument(PreprocessIntervals.LENGTH_OF_BINS_SHORT_NAME, Integer.toString(binLength))
        .addArgument(PreprocessIntervals.PADDING_SHORT_NAME, Integer.toString(padding))
        .addOutput(outputFile);
    inputIntervals.forEach(i -> argsBuilder.addArgument("L", i.getContig() + ":" + i.getStart() + "-" + i.getEnd()));
    runCommandLine(argsBuilder);

Note that most of our tests create the `String[]` of arguments as you've done here, but in cases like this where you need to add arguments dynamically, using the builder pattern is a nice option.

Reply:
Also, you can tell IntelliJ to "Optimize imports" once you make this change, as `import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;` will no longer be necessary.

Reply:
Did that, thanks for suggesting it.